modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -1475,94 +1475,155 @@ SUBROUTINE AllR16Ary5 (  Ary, AryDim1, AryDim2, AryDim3, AryDim4, AryDim5, Descr
+            ! Dont do this where performance matters.
+         ALLOCATE( TempArray( SIZE(Flags) + 1 ) )
+         DO J = 1, SIZE(Flags)
+            TempArray(J) = Flags(J)
+         END DO
+         TempArray(SIZE(Flags) + 1) = TRIM(Arg)
+         DEALLOCATE(Flags)
+         CALL MOVE_ALLOC(TempArray, Flags)
+      ELSE IF ( .NOT. FirstArgumentSet ) THEN
+         Arg1 = TRIM(Arg)
+         FirstArgumentSet = .TRUE.
+      ELSE IF ( .NOT. SecondArgumentSet ) THEN
+         Arg2 = TRIM(Arg)
+         SecondArgumentSet = .True.
+      ELSE
+         CALL INVALID_SYNTAX( 'too many command-line arguments given.' )
+         CALL CLEANUP()
+         RETURN
+      END IF
+   END DO
 
-         CALL GET_COMMAND_ARGUMENT( IArg, Arg, STATUS=Error )
+   DO I = 1, SIZE(Flags)
 
-         IF ( Error /= 0 )  THEN
-            CALL ProgAbort ( ' Error getting command-line argument #'//TRIM( Int2LStr( IArg ) )//'.', PRESENT(ErrStat) )
-            IF ( PRESENT(ErrStat) ) THEN
-               ErrStat = ErrID_Fatal
-               RETURN
-            END IF
-         END IF
+      Flag = Flags(I)(2:) ! This results in the flag without the switch character
+      CALL Conv2UC( Flag )
 
-         IF ( Arg(1:1) == SwChar .OR. Arg(1:1) == '-' ) THEN
-            IF (PRESENT(flag)) THEN
-               CALL Conv2UC( Arg )
-               Flag = Arg(2:) !this results in only the last flag
-               IF ( TRIM(Flag) == 'RESTART' ) CYCLE         ! Get next argument (which will be input [checkpoint] file name)
-            END IF
-                                                
-            CALL NWTC_DisplaySyntax( InputFile, ProgName )
+      SELECT CASE ( TRIM(Flag) )
 
-            IF ( INDEX( 'Hh?', Arg(2:2) ) > 0 ) THEN
-               IF ( PRESENT(ErrStat) ) THEN
-                  ErrStat = ErrID_Info !bjj? do we want to check if an input file was specified later?
-                  RETURN
-               ELSE
-                  CALL ProgExit ( 1 )
-               END IF
-            ELSE
-               CALL ProgAbort ( ' Invalid command-line switch "'//SwChar//TRIM( Arg(2:) )//'".', PRESENT(ErrStat) )
-               IF ( PRESENT(ErrStat) ) THEN
-                  ErrStat = ErrID_Fatal
-                  RETURN
-               END IF
-            END IF ! ( INDEX( 'Hh?', Arg(2:2)  ) > 0 )
-
-         ELSEIF ( FirstArg ) THEN
-            InputFile = Arg
-            FirstArg = .FALSE.
-         ELSE   
-            IF ( PRESENT(Arg2) ) THEN
-               Arg2 = Arg
-            END IF
-         END IF ! ( Arg(1:1) == SwChar )
+      CASE ('H')
+         CALL NWTC_DisplaySyntax( Arg1, ProgName )
+         CALL CLEANUP()
+         RETURN
 
-      END DO ! IArg
+      CASE ('RESTART')
+         IF ( .NOT. FirstArgumentSet .OR. .NOT. SecondArgumentSet ) THEN
+            CALL INVALID_SYNTAX( 'the restart capability requires and first and second argument: <input_file> -restart <checkpoint_file>' )
```

Comment:
Is there a reason that you are now requiring an extra argument for the restart capability?

The previous syntax was `<name of FAST executable > -restart <root name of checkpoint file without extension>`, i.e., `-restart <checkpoint_file>` with no need to supply `<input_file>`. The input file isn't used in restart, so I'd rather not have to supply an extra argument for it. The checkpoint file then *is* the input file.

Reply:
Fixed! Thanks for pointing that out.